### ISSUE TITLE
[question] Add Node (inclusion) fails to find new device, exits after 5 seconds

### ISSUE THREAD

**Issue by dbrandesky** [OpenZWave/Zwave2Mqtt#703]
Hi all, I'm fairly new to Z-Wave in general, so I might be missing something obvious here. So far, I've added two Jasco outdoor power outlets and one GE in-wall switch to my network with no trouble. Today, I tried to add a second GE in-wall switch of the same model as the first to my network, but every time I try to run "Add Node (inclusion)", it only searches for about 5 seconds; I immediately toggle the switch back and forth, but it never seems to get "discovered". The switch is in my detached garage, but the Aeon Z-Wave module is only about 20-30 feet away in my family room, not far from a big window that looks out on the garage, and the Jasco devices are even further away through multiple walls and a floor and still work no problem.

Here's what I see in OZW_log.txt:

```
2020-09-04 14:15:09.551 Detail, Queuing (Controller) Add Device
2020-09-04 14:15:09.555 Info, Add Device
2020-09-04 14:15:09.558 Detail, contrlr, Queuing (Command) ControllerCommand_AddDevice: 0x01, 0x05, 0x00, 0x4a, 0xc1, 0x21, 0x50
2020-09-04 14:15:09.561 Detail, Notification: AddDevice ControllerCommand - Starting
2020-09-04 14:15:09.564 Detail, 
2020-09-04 14:15:09.579 Info, contrlr, Sending (Command) message (Callback ID=0x21, Expected Reply=0x4a) - ControllerCommand_AddDevice: 0x01, 0x05, 0x00, 0x4a, 0xc1, 0x21, 0x50
2020-09-04 14:15:09.581 Info, contrlr, Encrypted Flag is 0
2020-09-04 14:15:09.586 Detail, contrlr,   Received: 0x01, 0x07, 0x00, 0x4a, 0x21, 0x01, 0x00, 0x00, 0x92
2020-09-04 14:15:09.589 Detail, 
2020-09-04 14:15:09.591 Info, contrlr, FUNC_ID_ZW_ADD_NODE_TO_NETWORK:
2020-09-04 14:15:09.592 Info, contrlr, ADD_NODE_STATUS_LEARN_READY
2020-09-04 14:15:09.593 Detail, contrlr,   Expected callbackId was received
2020-09-04 14:15:09.602 Detail, contrlr,   Expected reply was received
2020-09-04 14:15:09.604 Detail, contrlr,   Message transaction complete
2020-09-04 14:15:09.605 Detail, 
2020-09-04 14:15:09.606 Detail, contrlr, Removing current message
2020-09-04 14:15:09.607 Detail, Notification: AddDevice ControllerCommand - Waiting
2020-09-04 14:15:09.611 Info, WriteNextMsg Controller nothing to do
2020-09-04 14:15:14.556 Info, Cancel Add Node
2020-09-04 14:15:14.563 Detail, contrlr, Queuing (Command) Add Node Stop: 0x01, 0x04, 0x00, 0x4a, 0x05, 0xb4
2020-09-04 14:15:14.565 Detail, 
2020-09-04 14:15:14.568 Info, contrlr, Sending (Command) message (Callback ID=0x00, Expected Reply=0x4a) - Add Node Stop: 0x01, 0x04, 0x00, 0x4a, 0x05, 0xb4
2020-09-04 14:15:14.569 Info, contrlr, Encrypted Flag is 0
2020-09-04 14:15:14.577 Detail, contrlr, Notification: AddDevice ControllerCommand - Canceled
2020-09-04 14:15:14.661 Detail, contrlr,   Received: 0x01, 0x07, 0x00, 0x4a, 0x21, 0x06, 0x00, 0x00, 0x95
2020-09-04 14:15:14.666 Detail, 
2020-09-04 14:15:14.669 Info, contrlr, FUNC_ID_ZW_ADD_NODE_TO_NETWORK:
2020-09-04 14:15:14.675 Info, contrlr, ADD_NODE_STATUS_DONE
2020-09-04 14:15:14.676 Detail, contrlr,   Expected reply was received
2020-09-04 14:15:14.677 Detail, contrlr,   Message transaction complete
2020-09-04 14:15:14.680 Detail, 
2020-09-04 14:15:14.686 Detail, contrlr, Removing current message
2020-09-04 14:15:14.688 Detail, contrlr, Notification: AddDevice ControllerCommand - Completed
```

Not sure if that sheds any light. I have restarted the Zwave2Mqtt docker instance to see if that resolves the problem (early on I found if I ran too many different network operations it would just sort of hang up on me), and I also factory reset the new switch, to no avail.

Is there any other utility I can use to see if my Z-Wave controller is actually seeing any messages from this switch? The switch is in a plastic enclosure but there is a metal faceplate on the box; I don't think that should affect reception too much.

**Comment by robertsLando** [comment]
Based on your logs seems the inclusion is done successfully. I don't see any errors

**Comment by dbrandesky** [comment]
> Based on your logs seems the inclusion is done successfully. I don't see any errors

Very interesting, even if I select "show hidden nodes", I don't see any new devices. I see 3 deleted devices from my previous initial experimentation, and the "main controller", but that's it. Could there be some issue between the discovery of the device, and the actual adding of it to the node list?

**Comment by robertsLando** [comment]
Sorry I explained it wrongly, it seems that the controller command is received correctly, I don't see any device added, does the log stops here?

**Comment by dbrandesky** [comment]
@robertsLando yes, the log ends at `2020-09-04 14:15:14.688 Detail, contrlr, Notification: AddDevice ControllerCommand - Completed`

**Comment by robertsLando** [comment]
@dbrandesky DOes this happen only for that device? Are you able to add/remove others now?

**Comment by dbrandesky** [comment]
@robertsLando I don't have any other devices to add currently, but some new ones should be arriving in the next few days.

**Comment by robertsLando** [comment]
@dbrandesky Could you please try to exclude a device so and re add it back to see if you are able to do that?

I would like to check if it's a single device issue or a controller issue

**Comment by dbrandesky** [comment]
@robertsLando I just removed and re-added one of my Jasco outlets, worked just fine. 

Logs: [https://pastebin.com/WLZD2BWd](url)

**Comment by dbrandesky** [comment]
OK, so a GE Z Wave dimmer arrived today, so I gave that a shot, connecting it to a simple cheater power cord so I could attempt to add it from various places in case reception was bad in my garage. Even standing in the same room as my Zwave2MQTT box, I still couldn’t get it to connect to the new device. The logs show the same output as earlier.

FWIW I’m using a “branch” build of 3.4.1 that someone else posted here because the current master docket image wouldn’t run on raspberry pi. I’m on my phone right now otherwise I would link the relevant issues.

**Comment by dbrandesky** [comment]
> Coul you please try to use latest z2m version?

@robertsLando apparently, I cannot. I just tried, and got the same error I received before I switched to samsonnguyen's branch:
`dan_zwave2mqtt_1 exited with code 139`

I'm on a Raspberry Pi Zero W, so armv6.

**Comment by dbrandesky** [comment]
Not sure if this is relevant, but for whatever reason, I have to use docker-compose.yml `version 3` instead of the specified version 3.7. I haven't yet looked into what I need to do to get version 3.7 but I suspect it's because Debian is always way behind current application versions.

**Comment by robertsLando** [comment]
Actually I'm not on my laptop to check this but the docker image should be available for almost all archs (included armv6). I will be on holiday since next Monday, let me know if you find a way to fix this

**Comment by dbrandesky** [comment]
Well, I shut down my pi zero to try a better power supply, and when I went to turn it back on, it wouldn’t boot again; turns out the SD card data was corrupted. Luckily I backed up the config files beforehand, so I ended up rebuilding the whole setup using a raspberry pi 3B. I still couldn’t add the switch initially, but for some reason it just “showed up” several hours later. From there I was able to add another switch in the same area. I’m guessing there was some interference in the Z wave signal, and when I powered up more Z wave devices the network rebuilt and the signal was repeated back to the main controller. So, problem solved?
